pkg/registry/batch/job/strategy.go

```diff
@@ -130,6 +159,11 @@ func (jobStrategy) PrepareForUpdate(ctx context.Context, obj, old runtime.Object
 			newJob.Spec.Suspend = oldJob.Spec.Suspend
 		}
 	}
+	if !hasJobTrackingAnnotation(oldJob) {
+		// An existing Job can't be migrated from legacy tracking to tracking with
+		// finalizers.
+		dropJobTrackingAnnotation(newJob)
+	}
 
```

Comment:
what happens if the annotation gets deleted on update by the client making the API request? does the controller go back to tracking by pod counts? does that work properly (or at least no worse than job completion calculation currently works)? is it tested?